src/format/Token.cc

```diff
@@ -285,6 +287,66 @@ Format::Token::scanForToken(TokenTableEntry const table[], const char *cur)
     return cur;
 }
 
+// TODO: Reduce code duplication across this and other custom integer parsers.
+/// interprets input as an unsigned decimal integer that fits the specified Integer type
+template <typename Integer>
+static Integer
+ParseUnsignedDecimalInteger(const char *description, const SBuf &rawInput)
+{
+    constexpr auto minValue = std::numeric_limits<Integer>::min();
+    constexpr auto maxValue = std::numeric_limits<Integer>::max();
+
+    Parser::Tokenizer tok(rawInput);
+    if (tok.skip('0')) {
+        if (!tok.atEnd()) {
+            // e.g., 077, 0xFF, 0b101, or 0.1
+            throw TextException(ToSBuf("Malformed ", description,
+                                       ": Expected a decimal integer without leading zeros but got '",
+                                       rawInput, "'"), Here());
+        }
+        // for simplicity, we currently assume that zero is always in range
+        static_assert(minValue <= 0);
+        static_assert(0 <= maxValue);
```

Comment:
Why are these static_asserts here and not where minValue and maxValue are defined?

Reply:
> Why are these static_asserts here and not where minValue and maxValue are defined?

These static assertions are here because the `return 0` statement they are attached to is the only code that _uses_ the two asserted conditions. If you look closely, that `return` statement _bypasses_ the numeric_limits-based range checks further below. Imagine a hypothetical custom PositiveInteger type with std::numeric_limits<PositiveInteger>::min() returning 1. Without these static assertions, this function would either return 0 (as if zero were a valid PositiveInteger) or hit a runtime assertion in PositiveInteger constructor, violating its API promises in either case.

If we move these assertions elsewhere, it would be (even more) difficult to explain them IMO and, perhaps more importantly, it would be very difficult to remember to remove them if/when they become unnecessary in future code refactoring. In general, we should place code-specific assertions in that code (where possible).